api/v1beta1/infraenv_types.go

```diff
@@ -114,9 +115,15 @@ type InfraEnvSpec struct {
 	// OSImageVersion is the version of OS image to use when generating the InfraEnv.
 	// The version should refer to an OSImage specified in the AgentServiceConfig
 	// (i.e. OSImageVersion should equal to an OpenshiftVersion in OSImages list).
-	// Note: OSImageVersion can't be specified along with ClusterRef. 
+	// Note: OSImageVersion can't be specified along with ClusterRef.
 	// +optional
 	OSImageVersion string `json:"osImageVersion,omitempty"`
+
+	// MirrorRegistryRef is a reference to ClusterMirrorRegistry ConfigMap that holds the registries toml
+	// data
+	// Set per cluster mirror registry
+	// +optional
+	MirrorRegistryRef *v1beta1.MirrorRegistryConfigMapReference `json:"mirrorRegistryRef,omitempty"`
```

Comment:
What do you think about using the `corev1.ObjectReference`? Similar to https://github.com/openshift-assisted/cluster-api-agent/blob/9abff0c472c9156bde1cbc936fcc5bb03db7700f/bootstrap/api/v1alpha1/agentbootstrapconfig_types.go#L85

Reply:
I though so too but after reading `corev1.ObjectReference` documentation I decided to create a new reference type. LMK if still you think we should use `corev1.ObjectReference` 

``` go
// Instead of using this type, create a locally provided and used type that is well-focused on your reference.
// For example, ServiceReferences for admission registration: https://github.com/kubernetes/api/blob/release-1.17/admissionregistration/v1/types.go#L533 .
```